eng/common/TestResources/New-TestResources.ps1

```diff
@@ -52,6 +52,9 @@ param (
     [ValidateRange(1, [int]::MaxValue)]
     [int] $DeleteAfterHours = 120,
 
+    [Parameter()]
+    [switch] $DoNotApplyDeleteAfterTag,
```

Comment:
This is not powershelly. Why not just set `DeleteAfterHours` to 0 and add a condition to not set it if 0 (or less, since it's an int since uint params are bad...or add a `[ValidateInRange(0, [int]::MaxValue)]` parameter. More parameters just complicates the story.